app/lib/state_change_notifier.rb

```diff
@@ -86,12 +89,23 @@ def self.disable_notifications
   def providers_for(event, applications)
     return [] if event == :recruited
 
-    applications.select(&"#{event}?".to_sym).map(&:provider).map(&:name)
+    applications[event].map(&:provider).map(&:name)
   end
 
   def text_for(applications, event)
     providers = providers_for(event, applications)
+    return if providers.none?
+
+    I18n.t("slack_notifications.#{event}.message.other", providers: providers.to_sentence, count: providers.count)
+  end
+
+  def group_by_status(applications)
+    applications.inject(APPLICATION_OUTCOME_EVENTS.index_with { |_| [] }) do |grouped, application|
```

Comment:
It seems that `index_with([])` attempts to update the existing array rather than a new instance of it so I cannot make the change as I'm getting a `FrozenError` when I attempt to make this change.